<launch>
  <!-- Global params for all rules and nodes -->
  <param name="number_robots" value="10" />
  <param name="refresh_rate" value="10" />

  <!-- stage -->
  <node pkg="sphero_stage" type="start.py" name="sphero_stage" required="true" />

  <!-- Principal Node. Head of the system -->
  <node pkg="reynolds_rules" type="ReynoldsRulesNode.py" name="reynolds_rules_node" required="true" output="screen" >
    <rosparam>
      separation_weight: 0.8
      alignment_weight: 0.5
      nav2point_weight: 0.1
      cohesion_weight: 0.1
      obstacle_avoidance_weight: 0.8
      threshold_priorities: 1
    </rosparam>
  </node>

  <!-- Now run every other node (rule) with neccessary params or remaps -->
  <node pkg="reynolds_rules" type="SeparationRuleNode.py" name="separation_rule">
    <rosparam>
      view_range: 0.3
    </rosparam>
  </node>

  <node pkg="reynolds_rules" type="AlignmentRuleNode.py" name="alignment_rule" />

  <node pkg="reynolds_rules" type="CohesionRuleNode.py" name="cohesion_rule">
    <rosparam>
      view_range: 0.3
    </rosparam>
  </node>

  <node pkg="reynolds_rules" type="ObstacleAvoidanceNode.py" name="obstacle_avoidance_rule">
    <rosparam>
      view_range: 0.5
      view_angle: 60
      view_split: 0
    </rosparam>
  </node>

  <node pkg="reynolds_rules" type="Nav2PointRuleNode.py" name="nav2point_rule">
    <rosparam>
      point_x : 4
      point_y : 0
    </rosparam>
  </node>

  <!-- stub for the map server -->
  <node pkg="reynolds_rules" type="MapStubNode.py" name="map">
    <rosparam>
      map_file : "hard_maze_10x10.json"
    </rosparam>
  </node>
</launch>
